src/services/paras/ParasService.ts

```diff
@@ -102,4 +121,255 @@ export class ParasService extends AbstractService {
 			funds: entries,
 		};
 	}
+
+	/**
+	 * @param hash `BlockHash` to make call at
+	 * @param paraId ID of para to get lease info of
+	 * @returns current and future lease info as well as lifecycle information for
+	 * a given `paraId`
+	 */
+	async leaseInfo(hash: BlockHash, paraId: number): Promise<ILeaseInfo> {
+		const [leases, { number }, paraLifeCycle] = await Promise.all([
+			this.api.query.slots.leases.at<
+				Vec<Option<ITuple<[AccountId, BalanceOf]>>>
+			>(hash, paraId),
+			this.api.rpc.chain.getHeader(hash),
+			this.api.query.paras.paraLifecycles.at<ParaLifecycle>(hash, paraId),
+		]);
+		const blockNumber = number.unwrap();
+
+		const at = {
+			hash,
+			height: blockNumber.toString(10),
+		};
+
+		let leasesFormatted;
+		if (leases.length) {
+			const currentLeasePeriodIndex = this.currentLeasePeriodIndex(
+				blockNumber
+			).toNumber();
+
+			leasesFormatted = leases.reduce((acc, curLeaseOpt, idx) => {
+				if (curLeaseOpt.isSome) {
+					const leasePeriodIndex = currentLeasePeriodIndex + idx;
+					const lease = curLeaseOpt.unwrap();
+					acc.push({
+						leasePeriodIndex,
+						account: lease[0],
+						deposit: lease[1],
+					});
+				}
+
+				return acc;
+			}, [] as LeaseFormatted[]);
+		} else {
+			leasesFormatted = null;
+		}
+
+		let onboardingAs: ParaType | undefined;
```

Comment:
Could we make an `IOption<T>` for undefined's?

Reply:
I would be open to it, but would want to call it something else. 
My only hold up though are that `undefined` fn params and interface properties already have the `?` operator which is normally preferred over having a union type with undefined.